src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -0,0 +1,97 @@
+import com.google.protobuf.ByteString
+import io.grpc.Status
+import io.grpc.StatusException
+import java.util.Arrays
+import java.util.ArrayList
+
+/**
+ * Special class that can work with requests from CosmasClient
+ * This realization stores files in Google Cloud Storage.
+ *
+ * @author Aleksandr Fedotov (iisuslik43)
+ */
+class CosmasGoogleCloudService(private val bucketName: String) : CosmasGrpc.CosmasImplBase() {
+
+    private var storage: Storage = StorageOptions.getDefaultInstance().service
+
+    private constructor(bucketName: String, storage: Storage) : this(bucketName) {
+        this.storage = storage
+    }
+
+    override fun createVersion(request: CosmasProto.CreateVersionRequest,
+                               responseObserver: StreamObserver<CosmasProto.CreateVersionResponse>) {
+        println("Get request for create new version of file # ${request.fileId}")
+        try {
+            this.storage.create(
+                    BlobInfo.newBuilder(this.bucketName, request.fileId).build(),
+                    request.file.toByteArray())
+        } catch (e: StorageException) {
+            println("StorageException happened: ${e.message}")
+            responseObserver.onError(e)
+            return
+        }
+        val response: CosmasProto.CreateVersionResponse = CosmasProto.CreateVersionResponse
+                .newBuilder()
+                .build()
+        responseObserver.onNext(response)
+        responseObserver.onCompleted()
+    }
+
+    override fun getVersion(request: CosmasProto.GetVersionRequest,
+                            responseObserver: StreamObserver<CosmasProto.GetVersionResponse>) {
+        println("Get request for version ${request.version} file # ${request.fileId}")
+        val blob: Blob? = try {
+            this.storage.get(BlobInfo.newBuilder(this.bucketName, request.fileId).build().blobId)
+        } catch (e: StorageException) {
+            println("StorageException happened: ${e.message}")
+            responseObserver.onError(e)
+            return
+        }
+        val response = CosmasProto.GetVersionResponse.newBuilder()
+        if (blob != null) {
+            response.file = ByteString.copyFrom(blob.getContent())
+            responseObserver.onNext(response.build())
+            responseObserver.onCompleted()
+        } else {
+            val requestStatus = Status.NOT_FOUND.withDescription("There is no such file in storage")
+            println("This request is incorrect: " + requestStatus.description)
+            responseObserver.onError(StatusException(requestStatus))
+        }
+    }
+
+    fun deleteFile(fileId: String) {
+        println("Delete file # $fileId")
+        try {
+            this.storage.delete(BlobInfo.newBuilder(this.bucketName, fileId).build().blobId)
+        } catch (e: StorageException) {
+            println("Deleting file failed: ${e.message}")
+        }
+    }
+
+    companion object {
```

Comment:
... and if you move this stuff to the testing code you will actually have better separation between production and testing. You only need an object implementing `Storage` interface in this service implementation, you don't care if it comes from tests or from the real GCS

Reply:
Ok, thanks for your explanations, I deleted this private constructor